### PR TITLE
feat: Added support for creating a new PostgreSQL instance pointing to a PITR time

### PR DESCRIPTION
**Pull request by Ak-sky**
### Description

Support for creating a new PostgreSQL instance pointing to a PITR time

### Types of changes in this PR
https://github.com/terraform-ibm-modules/terraform-ibm-icd-postgresql/issues/44
#### No release required

- [x] Examples or tests (addition or updates of examples or tests)
- [x] Documentation update
- [ ] CI-related update (pipeline, etc.)
- [ ] Other changes that don't affect Terraform code

#### Release required

- [ ] Bug fix (patch release (`x.x.X`): Change that fixes an issue and is compatible with earlier versions)
- [x] New feature (minor release (`x.X.x`): Change that adds functionality and is compatible with earlier versions)
- [ ] Breaking change (major release (`X.x.x`): Change that is likely incompatible with previous versions)

##### Release notes content

Replace this text with information that users need to know about the bug fixes, features, and breaking changes. This information helps the merger write the commit message that is published in the release notes for the module.

---

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### Merge actions for mergers

- Merge by using "Squash and merge".
- Use a relevant [conventional commit](https://www.conventionalcommits.org/) message that is based on the PR contents and any release notes provided by the PR author.

    The commit message determines whether a new version of the module is needed, and if so, which semver increment to use (major, minor, or patch).

### CODE REVIEW

File: examples/pitr/main.tf [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+module "postgresql_db" {
+  source            = "../.."
+  resource_group_id = module.resource_group.resource_group_id
+  name              = "${var.prefix}-postgres"
+  region            = var.region
+  resource_tags     = var.resource_tags
+  configuration     = var.configuration
+}
+
+resource "null_resource" "wait_for_postgresql_db_pitr_creation" {
```
Comment (vburckhardt) [inline examples/pitr/main.tf:25]: could we use native tf support for the sleep here: https://registry.terraform.io/providers/hashicorp/time/latest/docs/resources/sleep

File: examples/pitr/main.tf [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  resource_group_name          = var.resource_group == null ? "${var.prefix}-resource-group" : null
+  existing_resource_group_name = var.resource_group
+}
+locals {
+  permanent_resources = yamldecode(file("../../common-dev-assets/common-go-assets/common-permanent-resources.yaml"))
+}
+
+data "ibm_database_point_in_time_recovery" "database_pitr" {
+  deployment_id = local.permanent_resources["postgresqlCrn"]
+}
```
Comment (ocofaigh) [inline examples/pitr/main.tf:17]: this should be done in the go test, not the example. The example just need to have a variable where any CRN can be passed so its not specific to our account

Comment (ocofaigh) [reply examples/pitr/main.tf:17]: see example https://github.com/terraform-ibm-modules/terraform-ibm-cos/blob/main/tests/pr_test.go#L26

Comment (Ak-sky) [reply examples/pitr/main.tf:14]: implemented.

**Review by ocofaigh — APPROVED** [review]

**Comment by ocofaigh** [comment]
Pipeline failed with `Error: [ERROR] Error getting user connection string for user (admin): [ERROR] Error getting database user connection string via ICD API: Request failed with status code: 404, ServerErrorResponse: {"errors":"not_found"}`

**Pull request merged**